design/trigger-backups-api-events.md

```diff
@@ -0,0 +1,35 @@
+# Trigger backups based on Kubernetes events 
+
+Triggering backups with Velero based on Kubernetes api events (Terminating, deleting, CrashLoopBackOff, etc..).
+
+## Goals
+
+- Rather than performing backups manually or by scheduling them, backups are created based on events.
+- Recovering the components of the cluster when they were deleted accidently. 
+- Blocking the delete process of a component in the cluster if it was a human intervention and trigger a backup.
+
+## Non Goals
+
+- N/A
+
+## Background
+
+## High-Level Design
+
+A custom controller that list and watch all the events of Kubenretes and based on that it trigger backups.
+
+## Detailed Design
+
+A custom controller where it have its own refelctor to list and watch a specific components using the kubernetes watch api. 
```

Comment:
How would users specify what components to watch? How would users specify what events to watch for?

Reply:
this could be achieved through namespace selectors. And for specifying what events to watch; this should be by default watching any delete events..

Reply:
Namespaces are one way, sure. I was also thinking about specifying per resource type.

I'd recommend providing explicit examples of what the user would type to list out events and namespaces. What would the configuration file or CRD look like? Would it be an addition to an existing CRD? These kinds of questions would be good to be answered in the proposal itself.